### ISSUE TITLE
[eslint-plugin-lit-a11y]: implement no-noninteractive-element-interactions

### ISSUE THREAD

**Issue by bennypowers** [open-wc/open-wc#1928]
See: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-noninteractive-element-interactions.md

# no-noninteractive-element-interactions

Non-interactive HTML elements and non-interactive ARIA roles indicate _content_ and _containers_ in the user interface. A non-interactive element does not support event handlers (mouse and key handlers). Non-interactive elements include `<main>`, `<area>`, `<h1>` (,`<h2>`, etc), `<p>`, `<img>`, `<li>`, `<ul>` and `<ol>`. Non-interactive [WAI-ARIA roles](https://www.w3.org/TR/wai-aria-1.1/#usage_intro) include `article`, `banner`, `complementary`, `img`, `listitem`, `main`, `region` and `tooltip`.

## Rule details

You may configure which handler props should be taken into account when applying this rule. The recommended configuration includes the following 6 handlers.

```javascript
'lit-a11y/no-noninteractive-element-interactions': ['error', {
    customElements: [
      'custom-input'
    ],
    events: [
      'click',
      'mousedown',
      'mouseup',
      'keypress',
      'keydown',
      'keyup',
    ],
  },
],
```

Adjust the list of event names in the events array to add custom events and increase or decrease the coverage surface of this rule in your codebase.

### Succeed
```js
html`
  <div @click=${() => void 0} role="button"></div>
  <div @click=${() => void 0} role="presentation"></div>
  <!-- Interactive element does not require role. -->
  <input type="text" @click=${() => void 0} /> 
  <button @click=${() => void 0} className="foo" /> 

  <!-- This is hidden from screenreader. -->
  <div @click=${() => void 0} role="button" aria-hidden="true"></div>

  <!-- custom elements listed in the options will be checked, other custom elements will be ignored -->
  <custom-input @click=${() => void 0} type="hidden"></custom-input>
`;
```

### Fail
```js
html`
  <li @click=${() => void 0}></li>
  <div @click=${() => void 0} role="listitem"></div>
`;
```

## Accessibility guidelines
- [WCAG 4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value)

### Resources
- [WAI-ARIA roles](https://www.w3.org/TR/wai-aria-1.1/#usage_intro)
- [WAI-ARIA Authoring Practices Guide - Design Patterns and Widgets](https://www.w3.org/TR/wai-aria-practices-1.1/#aria_ex)
- [Fundamental Keyboard Navigation Conventions](https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_generalnav)
- [Mozilla Developer Network - ARIA Techniques](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_button_role#Keyboard_and_focus)
